src/common/hook/usePublishQueue/index.tsx

```diff
@@ -0,0 +1,70 @@
+import { useState } from "react";
+import { FormEntry, Config } from "../../../types";
+import { getPublishingQueue } from "./utils/publish";
+import { getDefaultProvider } from "ethers";
+import { ContractFunctionState } from "@govtechsg/ethers-contract-hook";
+import { DocumentStoreFactory } from "@govtechsg/document-store";
+import { DocumentStore } from "@govtechsg/document-store/src/contracts/DocumentStore";
+
+interface WrappedDocument {
+  fileName: string;
+  data: any;
+}
+
+export const usePublishQueue = (
+  config: Config,
+  formEntries: FormEntry[]
+): {
+  error?: string;
+  publishState: string;
+  publish: () => void;
+  wrappedDocuments: WrappedDocument[];
+} => {
+  const [error, setError] = useState<string>();
+  const [publishState, setPublishState] = useState<ContractFunctionState>("UNINITIALIZED");
+  const [wrappedDocuments, setWrappedDocuments] = useState<WrappedDocument[]>([]);
+
+  config.wallet.connect(getDefaultProvider(config.network));
+
+  const publish = () => {
+    const publishQueue = getPublishingQueue(formEntries, config);
+    const nextWrappedDocuments: WrappedDocument[] = [];
+    publishQueue.forEach((queuedRequest) => {
```

Comment:
Refactor to use task list: 

[{
  type: "VERIFIABLE_DOCUMENT" | "TRANSFERABLE_RECORD",
  payload: {},
  documents: [{
    name: "document-1.tt",
    data: {}
  }, {
    name: "document-1.tt",
    data: {}
  }]
}]

When a `task` is completed, it gets pushed onto the completed task list and can be shown on the frontend.
This way, a document store task is processed the same way as a transferable record task which also requires a beneficiary and holder creation.